client/checkout/api/index.js

```diff
@@ -238,9 +264,15 @@ export default class WCPayAPI {
 							: 'false',
 					} );
 
-					return [ ajaxCall, result.error ];
+					return [ ajaxCall, paymentError || result.error ];
 				} )
-				.then( ( [ verificationCall, originalError ] ) => {
+				.then( ( res ) => {
+					const [
+						verificationCall,
+						paymentError,
+						originalError,
+					] = res;
```

Comment:
Looking at the Promise chain, is there a reason for destructuring three elements while the returned array consists of two only? Does this code mean we effectively moved throwing the `originalError` to after the veriication call, as `originalError` is in fact `paymentError` now?

Reply:
This was an attempt at throwing either error with the thinking that only one would ever be preset, but you're right! Thanks for catching it! We can just return all the errors. [e4ddcc0](https://github.com/Automattic/woocommerce-payments/pull/10442/commits/e4ddcc0e64cf41376a439116c80f0eea934a1cfc)